plugins/modules/provision_workflow_manager.py

```diff
@@ -1496,42 +1594,69 @@ def get_diff_deleted(self):
+                    family="sda",
+                    function="delete_provisioned_wired_device",
+                    op_modifies=True,
+                    params={
+                        "device_management_ip_address": self.validated_config["management_ip_address"]
+                    },
+                )
+                self.log("Response collected from the 'delete_provisioned_wired_device' API is : {0}".format(str(response)), "DEBUG")
 
-        else:
-            response = self.dnac._exec(
-                family="sda",
-                function='delete_provisioned_devices',
-                op_modifies=True,
-                params={'networkDeviceId': device_id},
-            )
-            self.log("Received API response from 'delete_provisioned_devices': {0}".format(str(response)), "DEBUG")
-            self.check_tasks_response_status(response, api_name='delete_provisioned_devices')
-            if self.status not in ["failed", "exited"]:
+                task_id = response.get("taskId")
+                deletion_info = self.get_task_status(task_id=task_id)
                 self.result["changed"] = True
-                self.result['msg'] = "Deletion done Successfully for the device '{0}' ".format(self.validated_config["management_ip_address"])
+                self.result['msg'] = "Deletion done Successfully"
                 self.result['diff'] = self.validated_config
-                self.result['response'] = self.result['msg']
+                self.result['response'] = task_id
                 self.log(self.result['msg'], "INFO")
                 return self
 
+            except Exception as e:
+                self.msg = "Error in delete provisioned device '{0}' due to {1}".format(self.device_ip, str(e))
+                self.log(self.msg, "ERROR")
+                self.status = "failed"
+                self.result['response'] = self.msg
+                self.check_return_status()
+
+        else:
+            try:
+                response = self.dnac._exec(
+                    family="sda",
+                    function='delete_provisioned_devices',
+                    op_modifies=True,
+                    params={'networkDeviceId': device_id},
+                )
+                self.log("Received API response from 'delete_provisioned_devices': {0}".format(str(response)), "DEBUG")
+                self.check_tasks_response_status(response, api_name='delete_provisioned_devices')
+
+                if self.status not in ["failed", "exited"]:
+                    self.result["changed"] = True
+                    self.result['msg'] = "Deletion done Successfully for the device '{0}' ".format(self.validated_config["management_ip_address"])
+                    self.result['diff'] = self.validated_config
+                    self.result['response'] = self.result['msg']
+                    self.log(self.result['msg'], "INFO")
+                    return self
+
+                if self.status == 'failed':
```

Comment:
Why do we check only for failed? What happens if status is exited? Do we print any message?

Reply:
fixed the code